### PR TITLE
added cygwin option for compilation

### PR DESCRIPTION
**Pull request by joejoseph08**
added cygwin option for compilation

Description
-----------
added cygwin option for compilation

Test Steps
-----------
NA

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS/Demo/Posix_GCC/.vscode/settings.json [hunk 1]
```diff
@@ -10,7 +10,15 @@
...
+            "path": "C:\\cygwin64\\bin\\bash.exe",
+            "args": ["--login", "-i"],
+            "env": {
+                "MSYSTEM": "Cygwin",
+                "CHERE_INVOKING": "1",
+            }
         }
     },
-    "terminal.integrated.defaultProfile.windows": "MSYS2",
+    "terminal.integrated.defaultProfile.windows": "Cygwin", //    "terminal.integrated.defaultProfile.windows": "MSYS2",
```
Comment (aggarg) [inline FreeRTOS/Demo/Posix_GCC/.vscode/settings.json:23]: Why this change? Would it break other MSYS2?

Comment (joejoseph08) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json:23]: Yes i believe so, Thanks for pointing out..
May be i will revert back this change..

Comment (joejoseph08) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json:23]: The main reason why I added cygwin is because I was getting error with MSYS2
Error was something like unable to find "sys/times.h" while compilation (I'm not sure if it was times.h or other header file).

But the Cygwin was perfectly compiling and working properly.

Comment (aggarg) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json:23]: Can you not select the default profile by running **Terminal: Select Default Profile** as described here - https://code.visualstudio.com/docs/terminal/profiles?

Comment (kstribrnAmzn) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json:23]: Dumb question: I realize this would modify the default profile to use the Cygwin profile instead of the MSYS2 profile by default and may cause a failure for MSYS2, but wouldn't this defaulting to MSYS2 also impact Cygwin compilation?

If so maybe a comment here and/or an update to the README is required for this demo to select the correct profile.

Comment (joejoseph08) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json:23]: @kstribrnAmzn , Yes correct, but my intention is to keep the 'default profile' modification only to my local copy (I wont be uploading it).
@aggarg , Please let us know if we have any other solution.

Comment (aggarg) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json:23]: @joejoseph08 Do you plan to update as you suggested?

Comment (joejoseph08) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json:23]: @aggarg , Yes sir
I will do it End of Today itself...
Appologize for the late response..

Comment (joejoseph08) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json]: @aggarg , I've completed the revert of my modification.

**Review by aggarg — APPROVED** [review]

**Review by tony-josi-aws — APPROVED** [review]

File: FreeRTOS/Demo/Posix_GCC/.vscode/settings.json (hunk 1 above)
Comment (aggarg) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json]: Merged your PR. Thank you for your contribution!

Comment (joejoseph08) [reply FreeRTOS/Demo/Posix_GCC/.vscode/settings.json]: > Merged your PR. Thank you for your contribution!

Thank you for your patient support too :)
